src/Button.tsx

```diff
@@ -0,0 +1,241 @@
+/** @jsx jsx */
+import { jsx } from "@emotion/core";
+import React, { ComponentProps } from "react";
+import * as colors from "./colors";
+import * as CSS from "csstype";
+import { base } from "./typography";
+
+// Types that could use some improvement:
+// * Don't allow `children` and `icon` to be missing
+// * Don't allow `children` when `FAB`
+//
+// I was able to get guarantees to work, but only with very cryptic errors. I
+// decided it'd be best, for the time being, to `throw` if we use things
+// incorrectly.
+interface Props
+  extends React.DetailedHTMLProps<
+    React.ButtonHTMLAttributes<HTMLButtonElement>,
+    HTMLButtonElement
+  > {
+  /**
+   * Treat this button as a floating action button
+   *
+   * You must include an `icon` prop and you must _not_ include a `children`
+   * prop for a floating action button.
+   */
+  fab?: boolean;
```

Comment:
Good question @cheapsteak . `fab` and `variant` are orthogonal. I prefer sticking with FAB because (with an obviously not statistically significant sample size of 2), I know what it means and that's what @adamatorres used in the Zeplin doc. FAB = float action button. That can be of the normal or simple variant.

![image](https://user-images.githubusercontent.com/5224118/61640802-f5fd3100-ac6b-11e9-9d1b-49671f26473a.png)

Reply:
A "Floating Action Button" at least to me surely _sounds_ like a variant of a "Button", and that it's orthogonal to the variant prop is imo a description of the outcome of API design (implementation detail) and imo not necessarily what makes intuitive sense  

Using boolean props instead of a `variant` enum prop would allow us to avoid the question of 'why is X a "variant" and Y not a "variant"?'  

Consumers (or space-kit itself) could make a `<Fab/>` component that's just `(...props) => <Button {...props} circular/>`

Reply:
Does it make sense that I think of "small"/"large"/"normal" as specifying a "size" variant of a Button, and "fab"/"circular" as specifying a "shape" variant?

Reply:
I think this is getting into bikeshed territory. We're talking about different words to describe the same things.

I don't think the fab prop is going to cause an issue; if you think this is a blocker for the PR; then let's jump on a call. Otherwise, let's just leave it.

Reply:
I think that `variant` makes sense for this. I can make a `variant` prop that takes `'circular' | undefined`

Reply:
I'm not advocating for `fab` or `size` to be renamed to "variant", my point is that a "variant" prop for a `<Button/>` suffers from a similar imprecision as the word "account" does in (organizations have accounts, users have accounts, which type of account "account" refers to is ambiguous)   

To escape ambiguity/imprecision, either no prop should be called "variant" (because every prop is a variant), or props should make explicit what they are variating (`userAccount`, `orgAccount`; `sizeVariant`, `shapeVariant`)